cdap-docs/reference-manual/source/http-restful-api/http-restful-api-v3/metrics.rst

```diff
@@ -98,114 +373,162 @@ The general form of a metrics request is::
 
-.. rubric:: Comments
-
-The scope must be either ``system`` for system metrics or ``user`` for user-defined metrics.
-
-System metrics are either Application metrics (about Applications and their Flows, Procedures, MapReduce, Workflows and Workers) or they are Data metrics (relating to Streams or Datasets).
 
-User metrics are always in the Application context.
+.. rubric:: Comments
 
-For example, to retrieve the number of input data objects (“events”) processed by a Flowlet named *splitter*,
-in the Flow *CountRandomFlow* of the Application *CountRandom*, over the last 5 seconds, you can issue an HTTP
-GET method::
+For example, to retrieve the number of input data objects (“events”) processed by the Flowlet named *splitter*,
+in the Flow *CountRandom* of the example application *CountRandom*, over the last 5 seconds, you can issue an HTTP
+POST method::
 
-  GET <base-url>/metrics/system/apps/CountRandom/flows/CountRandomFlow/flowlets/
-          splitter/process.events.processed?start=now-5s&count=5
+  POST <base-url>/metrics/query?context=namespace.default.app.CountRandom.flow.CountRandom.flowlet.
+          splitter&metric=system.process.events.processed&start=now-5s&count=5
 
 This returns a JSON response that has one entry for every second in the requested time interval. It will have
 values only for the times where the metric was actually emitted (shown here "pretty-printed")::
 
-  HTTP/1.1 200 OK
-  Content-Type: application/json
-  {"start":1382637108,"end":1382637112,"data":[
-  {"time":1382637108,"value":6868},
-  {"time":1382637109,"value":6895},
-  {"time":1382637110,"value":6856},
-  {"time":1382637111,"value":6816},
-  {"time":1382637112,"value":6765}]}
+  {"start":1382637108,"end":1382637112,"series":[
+    {"time":1382637108,"value":6868},
+    {"time":1382637109,"value":6895},
+    {"time":1382637110,"value":6856},
+    {"time":1382637111,"value":6816},
+    {"time":1382637112,"value":6765}]
+  }
+
+Each run of a flow is :ref:`identified by a run-ID <rest-program-runs>`. To retrieve the
+aggregate of events processed by the run of a flow, you can issue an HTTP POST method::
 
-Each run of a flow is identified by a run-ID. To retrieve the aggregate of events processed by the
-run of a flow, you can issue an HTTP GET method::
+  POST <base-url>/metrics/query?context=namespace.default.app.CountRandom.flow.CountRandom.run.
+    bca50436-9650-448e-9ab1-f1d186eb2285.flowlet.splitter&metric=system.process.events.processed&aggregate=true
 
-  GET <base-url>/metrics/system/apps/CountRandom/flows/CountRandomFlow/runs/13ac3a50-a435-49c8-a752-83b3c1e1b9a8/flowlets/
-          splitter/process.events.processed?aggregate=true
+which will return something similar to::
+
+  {"startTime":0,"endTime":0,"series":[{"metricName":"system.process.events.processed",
+   "grouping":{},"data":[{"time":0,"value":11188}]}]}
 
 If the run-ID is not specified, we aggregate the events processed for all the runs of this flow.
```

Comment:
If you want to tell how to get values agregated across mulitple contexts - you better tell that in general. E.g: 

The result timeseries will represent aggregated values for the context specified. Currently sum is used as aggregation function. E.g. if you query for system.process.events.processed metric for a flow (across all flowlets), since this metric was emitted on flowlet level the result values will be sum of those.